rcl_lifecycle/src/transition_map.c

```diff
@@ -55,6 +55,12 @@ rcl_lifecycle_transition_map_fini(
 {
   rcl_ret_t fcn_ret = RCL_RET_OK;
 
+  // free valid transitions for all states
+  for (unsigned int i = 0; i < transition_map->states_size; ++i) {
+      if (transition_map->states[i].valid_transitions != NULL) {
+          allocator->deallocate(transition_map->states[i].valid_transitions, allocator->state);
```

Comment:
I'm not sure if it will resolve the test issues, but we should probably set the address to `NULL` after deallocating:

```c
transition_map->states[i].valid_transitions = NULL;
```

Reply:
Good call.  I did that in 05a6e8a, along with fixing the minor style issue.  I'll run another batch of CI here to see if that made a difference.